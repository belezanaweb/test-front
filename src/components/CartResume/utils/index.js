export const normalizeCartData = data => {
  return {
    productsQuantity: data.items.length,
    subTotal: data.subTotal.toFixed(2).replace('.', ','),
    shipping: data.shippingTotal.toFixed(2).replace('.', ','),
    discount: data.discount.toFixed(2).replace('.', ','),
    total: data.total.toFixed(2).replace('.', ','),
  };
};
