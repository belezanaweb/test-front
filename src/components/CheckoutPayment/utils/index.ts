import * as Yup from 'yup'

const errors = {
  name: 'Insira um nome válido',
  code: 'Insira um código válido',
  number: 'Insira um número válido',
  expiration: 'Insira uma data válida',
}

export const initialValues = {
  creditCardName: '',
  creditCardNumber: '',
  creditCardExpiration: '',
  creditCardCode: '',
}

const regexOnlyNumbers = /[^\d.-]+/g

export const CheckoutPaymentSchema = Yup.object().shape({
  creditCardName: Yup.string()
    .min(2, errors.name)
    .matches(/^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi, errors.name)
    .required(errors.name),
  creditCardNumber: Yup.string()
    .min(2, errors.number)
    .test('len', errors.number, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 16
    })
    .required(errors.number),
  creditCardExpiration: Yup.string()
    .min(2, errors.name)
    .test('len', errors.expiration, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 4
    })
    .required(errors.expiration),
  creditCardCode: Yup.string()
    .matches(/^[0-9]+$/, errors.code)
    .test('len', errors.code, (val) => val.length === 3)
    .required(errors.code),
})
