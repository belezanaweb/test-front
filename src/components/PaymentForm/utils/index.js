export const formatCardNumber = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{4})(?=\d)/g, '$1 ')
    .substr(0, 19);
};

export const formatHolderName = value => {
  return value.replace(/[^a-zA-Z\u00C0-\u017F\s]/gu, '').substr(0, 60);
};

export const formatExpirationDate = value => {
  return value
    .replace(/\D/g, '')
    .replace(/(\d{2})(\d{0,4})/, '$1/$2')
    .substr(0, 7);
};

export const formatCVVCode = value => {
  return value.replace(/\D/g, '').substr(0, 3);
};

const validateCreditCard = cardNumber => {
  cardNumber = cardNumber.replace(/\s/g, '');

  if (cardNumber.length !== 16) {
    return false;
  }

  const digits = cardNumber.split('').map(Number);

  // Algoritmo de Luhn para validar o número do cartão
  let sum = 0;

  for (let i = 0; i < digits.length; i++) {
    let digit = digits[i];

    if (i % 2 === 0) {
      digit *= 2;

      if (digit > 9) {
        digit -= 9;
      }
    }

    sum += digit;
  }

  // O número do cartão é válido se a soma dos dígitos for um múltiplo de 10
  return sum % 10 === 0;
};

const isInvalidDate = dateString => {
  const today = new Date();
  const [month, year] = dateString.split('/');
  if (month === '00') {
    return true;
  }
  const userDate = new Date(Number(year), Number(month) - 1);

  return userDate < new Date(today.getFullYear(), today.getMonth());
};

export const validateForm = state => {
  const errors = {};

  if (!validateCreditCard(state.cardNumber)) {
    errors.cardNumber = 'insira um número de cartão válido';
  }

  if (!state.cardHolderName) {
    errors.cardHolderName = 'insira um nome válido';
  }

  if (state.expirationDate.length < 7 || isInvalidDate(state.expirationDate)) {
    errors.expirationDate = 'insira uma data válida';
  }

  if (state.cvv.length < 3) {
    errors.cvv = 'insira um cvv válido';
  }

  return errors;
};
