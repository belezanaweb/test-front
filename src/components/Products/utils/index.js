export const normalizeCartData = data => {
  return data.items.reduce((acc, curr) => {
    const formattedProduct = {
      imageUrl: curr.product.imageObjects[0].small,
      productName: curr.product.name,
      price: curr.product.priceSpecification.price,
      maxPrice: curr.product.priceSpecification.maxPrice,
    };

    return [...acc, formattedProduct];
  }, []);
};
