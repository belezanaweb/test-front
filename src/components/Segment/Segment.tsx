import styled from 'styled-components';

const Segment = styled.div`
  background-color: transparent;
  border: 1px solid ${({ theme }) => theme.colors.gray};
  border-radius: 3px;
  padding: 15px 13px;
`;

export default Segment;
