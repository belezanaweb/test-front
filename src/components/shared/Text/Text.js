const colors = {
  primary: '#000000',
  secondary: '#515151',
  emphasis: '#9222DC',
  error: '#DE1616',
};

const sizes = {
  small: '12px',
  regular: '14px',
  large: '16px',
  xlarge: '20px',
};

const weights = {
  bold: '700',
  regular: '400',
};

const decorations = {
  none: 'none',
  scratch: 'line-through',
};

const Text = props => {
  const {
    children,
    color = 'primary',
    size = 'regular',
    weight = 'regular',
    decoration = 'none',
    component: Component = 'p',
    className,
  } = props;

  const styles = {
    color: colors[color],
    fontSize: sizes[size],
    fontWeight: weights[weight],
    textDecoration: decorations[decoration],
  };

  return (
    <Component className={className} style={styles}>
      {children}
    </Component>
  );
};

export default Text;
