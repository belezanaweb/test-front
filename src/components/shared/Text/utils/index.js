export { colors, decorations, sizes, weights } from './cssTokens.js';
