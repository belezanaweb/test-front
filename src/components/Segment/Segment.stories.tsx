import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import theme, { Theming } from '../../theme';

import Box from '../Box';
import Text from '../Text';

import Segment from './Segment';

const Provider = (storyFn: () => React.ReactNode) => (
  <BrowserRouter>
    <Theming theme={theme}>
      {storyFn()}
    </Theming>
  </BrowserRouter>
);

export default {
  component: Segment,
  decorators: [Provider],
  title: 'Segment',
};

export const Default = () => (
  <Box css={{ display: 'block', padding: '20px' }}>
    <Segment>
      <Text>This is a segment</Text>
    </Segment>
  </Box>
);
