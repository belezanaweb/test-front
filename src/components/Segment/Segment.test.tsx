import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { provider } from '../../utils/testUtils';

import Text from '../Text';
import Segment from './Segment';

describe('<Segment />', () => {
  it('should render Segment', () => {
    const { container } = provider((
      <Segment>
        <Text>This is a segment</Text>
      </Segment>
    ), render);

    expect(container).toMatchSnapshot();
  });
});
