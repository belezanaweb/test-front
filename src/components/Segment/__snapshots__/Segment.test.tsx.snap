// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Segment /> should render Segment 1`] = `
<div>
  <div
    class="sc-AxirZ dHpXsK"
  >
    <span
      class="sc-AxjAm kPvYly"
    >
      This is a segment
    </span>
  </div>
</div>
`;
